scipy/integrate/_tanhsinh.py

```diff
@@ -760,3 +760,381 @@ def _tanhsinh_iv(f, a, b, log, maxfun, maxlevel, minlevel,
+
+            f(x: ndarray, *args) -> ndarray
+
+         where each element of ``x`` is a finite real and ``args`` is a tuple,
+         which may contain an arbitrary number of arrays that are broadcastable
+         with `x`. `f` must represent a smooth, positive, and monotone decreasing
+         function of `x`.
+    a, b : array_like
+        Real lower and upper limits of summed terms. Must be broadcastable.
+        Each element of `a` must be finite and less than the corresponding
+        element in `b`, but elements of `b` may be infinite.
+    step : array_like
+        Finite, positive, real step between summed terms. Must be broadcastable
+        with `a` and `b`.
+    args : tuple, optional
+        Additional positional arguments to be passed to `f`. Must be arrays
+        broadcastable with `a`, `b`, and `step`. If the callable to be summed
+        requires arguments that are not broadcastable with `a`, `b`, and `step`,
+        wrap that callable with `f`. See Examples.
+    log : bool, default: False
+        Setting to True indicates that `f` returns the log of the terms
+        and that `atol` and `rtol` are expressed as the logs of the absolute
+        and relative errors. In this case, the result object will contain the
+        log of the sum and error. This is useful for summands for which
+        numerical underflow or overflow would lead to inaccuracies.
+    maxterms : int, default: 2**32
+        The maximum number of terms to evaluate when summing directly. 
+        Additional function evaluations may be performed for input
+        validation and integral evaluation. 
+    atol, rtol : float, optional
+        Absolute termination tolerance (default: 0) and relative termination
+        tolerance (default: ``eps**0.5``, where ``eps`` is the precision of
+        the result dtype), respectively. Must be non-negative
+        and finite if `log` is False, and must be expressed as the log of a
+        non-negative and finite number if `log` is True.
+
+    Returns
+    -------
+    res : OptimizeResult
+        An instance of `scipy.optimize.OptimizeResult` with the following
+        attributes. (The descriptions are written as though the values will be
+        scalars; however, if `func` returns an array, the outputs will be
+
+        arrays of the same shape.)
+        success : bool
+            ``True`` when the algorithm terminated successfully (status ``0``).
+        status : int
+            An integer representing the exit status of the algorithm.
+            ``0`` : The algorithm converged to the specified tolerances.
+            ``-1`` : Element(s) of `a`, `b`, or `step` are invalid
+            ``-2`` : Numerical integration reached its iteration limit; the sum may be divergent.
+            ``-3`` : A non-finite value was encountered.
+        sum : float
+            An estimate of the sum.
+        error : float
+            An estimate of the absolute error.
```

Comment:
Should mention that the error estimate assumes the sum contains nonnegative terms.